internal/pipe/snapcraft/snapcraft.go

```diff
@@ -314,3 +334,29 @@ func push(ctx *context.Context, snap *artifact.Artifact) error {
 	ctx.Artifacts.Add(snap)
 	return nil
 }
+
+// walks the src, recreating dirs and hard-linking files
+func link(src, dest string, mode os.FileMode) error {
```

Comment:
this is copied from the `docker` pipe isn't it?

its ok, just wanted to note that linking thing may break in some cases (check #1544), eventually we'll give it a shot at copying the files instead... (which will then impact here as well)

Reply:
Indeed, this is the same code. I thought about taking it into a util package but I decided against it, just to keep things simple.

Do you think there is something that needs to be improved here for this PR?